### PR TITLE
Feature/af form

### LINKED ISSUE #1904: Query analytical framework

### LINKED ISSUE #1905: Save analytical framework
- All of the details for analytical framework

### LINKED ISSUE #1907: Show prompt when leaving the edit page (if there are unsaved changes)

### LINKED ISSUE #2020: Setup form for AnalysisFramework
The form should include:
- Metadata for analysis framework
- The primary tagging (sections)
- The secondary tagging (widgets)

Some known problems
1. clientId is required on query but optional in mutation (may need to write custom PartialForm type)
2.clientId is not required on analysisFramework node
3. previewImage doesn't work

### LINKED ISSUE #2032: Add Organization Search Select Input using GQL

### LINKED ISSUE #2033: Synchronize client typings with server for analysis framework

### LINKED ISSUE #2036: Show errors during save for Analytical Framework

### LINKED ISSUE #2038: Update "Subnavbar" to add features of "FullPageHeader"

### PR DESCRIPTION
**Pull request by tnagorra**
Merge after https://github.com/the-deep/client/pull/2031

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [x] permission checks
- [x] translations

### CODE REVIEW

**Review by AdityaKhatri — DISMISSED** [review]

File: app/views/AnalyticalFramework/FrameworkForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,443 @@
...
+            }
+        },
+        [
+            setError, validate, frameworkId,
+            primaryTaggingPristine, secondaryTaggingPristine,
+            updateAnalysisFramework, createAnalysisFramework,
+        ],
+    );
+
+    const errorWithoutTaggings = { ...error };
```
Comment (AdityaKhatri) [inline app/views/AnalyticalFramework/FrameworkForm/index.tsx]: Lets not do this.

File: app/views/AnalyticalFramework/index.tsx [hunk 2]
```diff
@@ -1,47 +1,25 @@
...
-    Tab,
-    TabList,
-    TabPanel,
     ButtonLikeLink,
 } from '@the-deep/deep-ui';
 
 import PreloadMessage from '#base/components/PreloadMessage';
-import FullPageHeader from '#components/FullPageHeader';
+import SubNavbar from '#components/SubNavbar';
+import NavbarContext from '#components/SubNavbar/context';
```
Comment (AdityaKhatri) [inline app/views/AnalyticalFramework/index.tsx]:
```suggestion
import SubNavbarContext from '#components/SubNavbar/context';
```

File: app/components/SubNavbar/context.ts [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
+    iconsNode?: Element | null | undefined;
+    actionsNode?: Element | null | undefined;
+    childrenNode?: Element | null | undefined;
+
+    setIconsNode?: React.Dispatch<React.SetStateAction<Element | null | undefined>>;
+    setActionsNode?: React.Dispatch<React.SetStateAction<Element | null | undefined>>;
+    setChildrenNode?: React.Dispatch<React.SetStateAction<Element | null | undefined>>;
 }
 
 const NavbarContext = createContext<NavbarContextInterface>({
```
Comment (AdityaKhatri) [inline app/components/SubNavbar/context.ts]: Rename to SubNavbarContextInterface

File: app/views/AnalyticalFramework/FrameworkForm/index.tsx (hunk 1 above)
Comment (tnagorra) [reply app/views/AnalyticalFramework/FrameworkForm/index.tsx]: Do what?

**Review by AdityaKhatri — DISMISSED** [review]

**Review by AdityaKhatri — APPROVED** [review]

**Pull request merged**